File: clients/android/app/src/main/kotlin/com/lextures/android/core/lms/LmsFeatureModelsReading.kt
package com.lextures.android.core.lms

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class LibraryBook(
    val id: String,
    val orgId: String,
    val title: String,
    val author: String? = null,
    val coverUrl: String? = null,
    val lexileLevel: Int? = null,
    val fpBand: String? = null,
    val gradeBand: String? = null,
    val summary: String? = null,
)

@Serializable
data class LibraryBooksResponse(
    val books: List<LibraryBook>? = null,
)

@Serializable
data class ReadingLogEntry(
    val id: String,
    val bookId: String? = null,
    val bookTitle: String? = null,
    val logDate: String,
    val pagesRead: Int? = null,
    val reflection: String? = null,
    val loggedAt: String? = null,
)

@Serializable
data class ReadingLogListResponse(
    val entries: List<ReadingLogEntry>? = null,
)

@Serializable
data class PostReadingLogBody(
    val bookId: String? = null,
    val bookTitle: String? = null,
    val logDate: String,
    val pagesRead: Int? = null,
    val reflection: String? = null,
)

@Serializable
data class PostReadingLogResponse(
    val entry: ReadingLogEntry,
)

File: clients/ios/Lextures/Core/LMS/LMSFeatureModelsReading.swift
import Foundation

// MARK: - Reading log & leveled library (M8.4)

struct LibraryBook: Codable, Identifiable, Hashable {
    var id: String
    var orgId: String
    var title: String
    var author: String?
    var coverUrl: String?
    var lexileLevel: Int?
    var fpBand: String?
    var gradeBand: String?
    var summary: String?
}

struct LibraryBooksFilter {
    var lexileMin: Int?
    var lexileMax: Int?
    var gradeBand: String?
}

struct LibraryBooksResponse: Decodable {
    var books: [LibraryBook]?
}

struct ReadingLogEntry: Codable, Identifiable, Hashable {
    var id: String
    var bookId: String?
    var bookTitle: String?
    var logDate: String
    var pagesRead: Int?
    var reflection: String?
    var loggedAt: String?
}

struct ReadingLogListResponse: Decodable {
    var entries: [ReadingLogEntry]?
}

struct PostReadingLogBody: Encodable {
    var bookId: String?
    var bookTitle: String?
    var logDate: String
    var pagesRead: Int?
    var reflection: String?
}

struct PostReadingLogResponse: Decodable {
    var entry: ReadingLogEntry
}

